OpenRA.Mods.RA/Activities/Teleport.cs

```diff
@@ -53,16 +55,59 @@ public override Activity Tick(Actor self)
 				if (condition.PreventsTeleport(self))
 					return NextActivity;
 
-			var bestCell = ChooseBestDestinationCell(self, destination);
-			if (bestCell == null)
-				return NextActivity;
+			if (!force)
+			{
+				var bestCell = ChooseBestDestinationCell(self, destination);
+				if (bestCell == null)
+					return NextActivity;
 
-			destination = bestCell.Value;
+				destination = bestCell.Value;
+			}
 
 			Game.Sound.Play(sound, self.CenterPosition);
 			Game.Sound.Play(sound, self.World.Map.CenterOfCell(destination));
 
 			self.Trait<IPositionable>().SetPosition(self, destination);
+			if (force)
+			{
+				if (self.Info.HasTraitInfo<HealthInfo>())
+				{
+					var health = self.Trait<Health>();
```

Comment:
Is there a reason why you didn't use `TraitOrDefault` and did a `null`-check above (instead of `HasTraitInfo`)?